Source/SpatialGDK/Private/SpatialActorChannel.cpp

```diff
@@ -364,42 +365,45 @@ bool USpatialActorChannel::ReplicateActor()
 	{		
 		if (RepFlags.bNetInitial && bCreatingNewEntity)
 		{
-			// When a player is connected, a FUniqueNetIdRepl is created with the players worker ID. This eventually gets stored
-			// inside APlayerState::UniqueId when UWorld::SpawnPlayActor is called. If this actor channel is managing a pawn or a 
-			// player controller, get the player state.
-			FString PlayerWorkerId;
-			APlayerState* PlayerState = Cast<APlayerState>(Actor);
-			if (!PlayerState)
+			if (!Actor->IsFullNameStableForNetworking() || SpatialNetDriver->GetSpatialInterop()->CanSpawnReplicatedStablyNamedActors())
 			{
-				APawn* Pawn = Cast<APawn>(Actor);
-				if (Pawn)
+				// When a player is connected, a FUniqueNetIdRepl is created with the players worker ID. This eventually gets stored
+				// inside APlayerState::UniqueId when UWorld::SpawnPlayActor is called. If this actor channel is managing a pawn or a 
+				// player controller, get the player state.
+				FString PlayerWorkerId;
+				APlayerState* PlayerState = Cast<APlayerState>(Actor);
+				if (!PlayerState)
 				{
-					PlayerState = Pawn->PlayerState;
```

Comment:
Drive-by, please move Pawn declaration into if statement.

Reply:
Fixed, there is also a ```APlayerState* PlayerState = Cast<APlayerState>(Actor);
if (!PlayerState)```
above, but that one has actual logic around that usage, so I will keep it.